inluxclient/client_query.go

```diff
@@ -0,0 +1,68 @@
+// Copyright 2021 InfluxData, Inc. All rights reserved.
+// Use of this source code is governed by MIT
+// license that can be found in the LICENSE file.
+
+package influxclient
+
+import (
+	"bytes"
+	"context"
+	"encoding/json"
+	"net/url"
+)
+
+// Dialect defines model for Dialect.
+type Dialect struct {
+	Annotations []string `json:"annotations"`
+	Delimiter   string   `json:"delimiter"`
+	Header      bool     `json:"header"`
+}
+
+// QueryParam is key-value pair for parameters inside the Flux query
+type QueryParam struct {
+	Key   string
+	Value string
+}
+
+// Query defines model for Query.
+type Query struct {
+	Dialect Dialect           `json:"dialect"`
+	Query   string            `json:"query"`
+	Type    string            `json:"type"`
+	Params  map[string]string `json:"params,omitempty"`
+}
+
+// defaultDialect is Query dialect value with all annotations, name header and comma as delimiter
+var defaultDialect = Dialect{
+	Annotations: []string{"datatype", "default", "group"},
+	Delimiter:   ",",
+	Header:      true,
+}
+
+// Query sends the given Flux query on the given organization ID.
+// The result must be closed after use.
+// Flux query can contain reference to params that must be passed in the params argument
+func (c *Client) Query(ctx context.Context, org, query string, params ...QueryParam) (*QueryResultReader, error) {
```

Comment:
I'm wondering if there's a good use case for having a client talking to many different organization IDs. ISTM that almost every realistic use case would involve the client talking to a single organization, so perhaps we could move the org ID to the client parameters, meaning that we wouldn't need to have it in every API endpoint? WDYT?

Another possible thought: instead of defining a `QueryParam` struct, we could accept a slice of alternating key-value strings, like strings.NewReplacer](https://pkg.go.dev/strings#NewReplacer) does, or maybe just a `map[string]string` ?

Reply:
The slice of alternating key-value slice strings is a very nice new pattern for me 👍. I like this more than the map because it allows using variadic function.

Reply:
Regarding OrgID, interesting idea. If we have OrgID as top-level argument it will simplify many future API functions.
Indeed, most users use the client for a single organization. I've discussed this with colleagues, authors of the other official clients. One of them mentioned a customer that  uses C# client to connect to multiple organizations, but it seems it is a minor use-case.

Reply:
Yeah, I think if someone wants to talk to multiple orgs, it wouldn't be too much hassle to have multiple clients, or create a client for each request.